lib/rex/proto/mssql/client.rb

```diff
@@ -83,6 +83,8 @@ def map_compile_os_to_platform(server_info)
             platform = Msf::Platform::Windows.realname
           elsif os_data.match?('win')
             platform = Msf::Platform::Windows.realname
+          elsif os_data.match?(/\bon\b\s+(\w+)/)
+            platform = os_data.match(/\bon\b\s+(\w+)/)[1]
```

Comment:
More curiosity than anything, but is `\b` more accurate/better (not sure what word to use here) than `\w`?

Reply:
I think it's just preference frankly, both would work I imagine. I'm no regex master, but I think you'd have to add the empty spaces to the string using `\w` - if you think that works better I'm happy to switch it up